[#12] Add repo_config and Deploy Info task

tmferretti opened the pull request.

tmferretti commented on lib/github_workflow/cli.rb:
```diff
@@ -460,6 +474,82 @@ def failure(message)
+          diff                     = `heroku pipelines:diff -r staging`
+          options[:commit_range]   = diff.split("\n").last.split("/").last
+          notes                    = formatted_deploy_notes(repo: name)
+          deploy_information[name] = {
+            diff:  diff,
+            notes: notes
+          }
```
this will get used during the `post_to_slack` method

tmferretti commented on lib/github_workflow/cli.rb:
```diff
@@ -460,6 +474,82 @@ def failure(message)
+      def post_to_slack
+        say_info("Post the output to Slack? y/n:")
+        if gets.chomp.downcase == "y"
+          slack_client.post(repo_config["webhook_url"], deploy_information_body)
```
all thats left on this pr is to build this method: `deploy_information_body`

ChrisCPO commented on lib/github_workflow/cli.rb:
```diff
@@ -22,6 +22,19 @@ class Cli < Thor
+          "auth-inbox":   "path/to/your/local/auth-inbox",
+          "ellis":        "path/to/your/local/ellis",
+          "fee-schedule": "path/to/your/local/fee_schedule",
+          "knowledgable": "path/to/your/local/knowledgable"
```
we should probably add wcb app here.

ChrisCPO replied:
@BrianBorge brought a good point, this is an open source. This here at least should look for a list from a config file... actually this gem does look at the `.github_workflow`  file sot these should be moved there, but that is app specific.

 
 However we are introducing slack and heroku in this PR. maybe they should be plugins, or this is not the best spot to do this. I'm fine with appointed software... but idkM
 
 commenting all over a DRAFT pr.... sorry

tmferretti replied:
no worries @ChrisCPO i noodled on this a bit. because without some single place for this dotfile, the cli wouldnt know where to chdir to to then pull down. but moving this out of gw would be a bigger lift because it heavily uses existing methods in gw

tmferretti replied:
also I dont see @BrianBorge's comment

ChrisCPO replied:
duplicating the list per repo would suck.. perhaps there needs to be a global dotfile/config file. `.github_workflow_config`

tmferretti commented:
BTW this cli is a bit of a mess. its largely all in one file

BrianBorge commented:
Neat, love seeing improvements to our deploy process.  Could you add a problem statement to the draft PR?

Before diving into the implementation details, some initial thoughts:

- I am unsure this should live in GH Workflow as it's an open-source project and knows nothing about repos specific to an organization (I realize [this is not entirely true](https://github.com/daisybill/github_workflow/blame/01874c1ea240f00281ea704fc2994a8dcfb751a4/lib/github_workflow/cli.rb#L436)).  It also introduces two new dependencies (Heroku and Slack).
- If we are considering adding references to daiysbill repos, we should consider close sourcing this or changing the code so it doesn't know about them.  And adding a short, "dependencies" block to the readme.
- I'm confused about how the Slack integration works.  Why is the class called `SlackBlockkitComponent`? (missing dependency?)

tmferretti commented:
> Neat, love seeing improvements to our deploy process. Could you add a problem statement to the draft PR?
> 
> Before diving into the implementation details, some initial thoughts:
> 
> * I am unsure this should live in GH Workflow as it's an open-source project and knows nothing about repos specific to an organization (I realize [this is not entirely true](https://github.com/daisybill/github_workflow/blame/01874c1ea240f00281ea704fc2994a8dcfb751a4/lib/github_workflow/cli.rb#L436)).  It also introduces two new dependencies (Heroku and Slack).
> * If we are considering adding references to daiysbill repos, we should consider close sourcing this or changing the code so it doesn't know about them.  And adding a short, "dependencies" block to the readme.
> * I'm confused about how the Slack integration works.  Why is the class called `SlackBlockkitComponent`? (missing dependency?)

As for 1. I agree, but its already org specific & there was a fair bit of free utility I used to build what I added (`init_fresh_main` and the `deploy_notes` for example) but i was trying to get out a prototype before you became the deploy manager.

2. Fair, I agree, see 1.

3. The slack integration is nothing fancy, just a simple webhook url (which contains a per user/channel secret!) that we make a POST to. the block kit component was a PORO that I added to abstract building the payload out of the main file. Id pull alot more of this file into PORO's but that was outside the scope of what I was trying to do.

tmferretti commented:
@BrianBorge @ChrisCPO this is the slack side of the integration: https://daisybill.slack.com/apps/A02AFMHR96E-daisyshepherd?next_id=0

tmferretti commented:
@benliscio, Do you mind If I close this PR? Hope you and the crew have been well.

BrianBorge commented:
@tmferretti - You're good to close it.
